Rom app improve fomat rom parameters

Rbravo555 opened the pull request:
**📝 Description**
This PR seeks to move the ROM information (namely the **nodal_modes** and the **elements_and_weights**) currently stored inside the RomParameters.json into 4 (or 5 if Petrov-Galerkin is used) numpy arrays: 
- RightBasisMatrix.npy
- LeftBasisMatrix.npy (if Petrov Galerkin is used)
- NodeIds.npy
- ElementsVector.npy
- WeightsMatrix.npy

**🆕 Changelog**
- In **calculate_rom_basis_output_process** store ModesMatrix.npy and NodeIds.npy to disk rather than appending them to the RomParameters.json
- In **hrom_training_utility** store WeightsMatrix.npy and ElementsVector.npy rather than appending them to the RomParameters
- In **rom_analysis** take the ROM data from the numpy arrays instead of the RomParameters.json

As dicussed a number of numpy arrays are stored to disk, rather than a single one that might be less transparent. The branch is fully compatible with Galerkin, LSPG and Petrov-Galerkin projections. Moreover, we have kept the option of using JSON for the ROM information if the user so decides

Rbravo555 committed:
Storing modes and node ids as Numpy

Rbravo555 committed:
Accepting modes and node ids as Numpy arrays

Rbravo555 committed:
Use Numpy for HROM elements and weights

Still need to update the functions calculating the include_minimum_condition and  include_condition_parents

Rbravo555 committed:
typo

Rbravo555 committed:
Update name of numpy files for consistency

Rbravo555 committed:
Adapting utilities for adding parent elements to Numpy format

Rbravo555 committed:
update comment

Rbravo555 committed:
Updating HROM parts creation to Numpy

Rbravo555 committed:
Updating comments

Rbravo555 committed:
Cleaning

Rbravo555 committed:
Erase commented code

Rbravo555 committed:
typo

Rbravo555 committed:
little bug

Rbravo555 committed:
Fix bug when creating HROM parts. 

Encoding a dictionary to convert it into Kratos Parameters

RiccardoRossi commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -82,9 +82,10 @@ def CreateHRomModelParts(self):
-            rom_parameters = KratosMultiphysics.Parameters(f.read())
-        KratosROM.RomAuxiliaryUtilities.SetHRomComputingModelPart(rom_parameters["elements_and_weights"], computing_model_part, hrom_main_model_part)
+        KratosROM.RomAuxiliaryUtilities.SetHRomComputingModelPart(
+            KratosMultiphysics.Parameters(json.JSONEncoder().encode(self.__CreateDictionaryWithRomElementsAndWeights())),
```
i am not clear about what you are doing here. could u expand a bit?

RiccardoRossi commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -192,19 +178,60 @@ def __AppendHRomWeightsToRomParameters(self):
+        return updated_weights, updated_conditions
+
+
+    def __CreateDictionaryWithRomElementsAndWeights(self, w = None, z=None, n_elements = None):
```
does it make sense to still call this "CreateDictionary"?

rubenzorrilla replied:
I think it does, it returns a dictionary.

RiccardoRossi commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -192,19 +178,60 @@ def __AppendHRomWeightsToRomParameters(self):
+        else:
+            # Many elements found
+            for j in range (0,len(z)):
+                if z[j] <=  n_elements -1:
```
i think that here you could do this by vectorized ops using numpy.

the thing that worries me a bit is if we are sure that this is not the part that takes long

RiccardoRossi commented on applications/RomApplication/python_scripts/rom_analysis.py:
```diff
@@ -100,13 +101,20 @@ def ModifyAfterSolverInitialize(self):
+                OriginalNumberOfElements = computing_model_part.NumberOfElements()
+                for i in range(WeightsMatrix.shape[0]):
+                    if ElementsVector[i] < OriginalNumberOfElements:
+                        computing_model_part.GetElement(int( ElementsVector[i])+1).SetValue(KratosROM.HROM_WEIGHT, WeightsMatrix[i]  )
```
if you want efficiency this needs to be in c++...

Rbravo555 committed:
Merge branch 'master' into RomApp_ImproveFomatRomParameters

Rbravo555 committed:
Adding Rom data in Numpy to all tests

Rbravo555 committed:
Using Nympy version of PetrovGalerkin modes

Rbravo555 committed:
Setting name of modes as Right- and LeftBasisMatrix

Rbravo555 committed:
Correcting Potential flow test

There used to be more nodal modes than nodes. The number of nodal modes was limited to 22, which is the number of nodes in simple_airfoil.mdpa

Rbravo555 committed:
Adding possibility to create numpy or json

Rbravo555 committed:
Single-mode case taken into account

Rbravo555 committed:
retrieving original number of elements from RomParameters.json

Useful for HROM model parts with only the hyper-reduced set of elements and conditions

Rbravo555 committed:
adding flag for numpy in RomParameters

Rbravo555 committed:
Making mandatory name:RomParameters.json

Potentially useful standard for RomManager

Rbravo555 committed:
Updating RomParameters.json to include "rom_format":"numpy"

Rbravo555 committed:
erasing possibility to change name of RomParameters.json

Rbravo555 committed:
Adding and accepting flag "rom_format" to RomParameters.json

Rbravo555 committed:
clean

NicolasSR commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -206,19 +195,67 @@ def __AppendHRomWeightsToRomParameters(self):
+        elif self.hrom_output_format=="json":
+            with open('RomParameters.json','r') as f:
+                updated_rom_parameters = json.load(f)
+                updated_rom_parameters["elements_and_weights"] = hrom_weights #TODO: Rename elements_and_weights to hrom_weights
```
In the json format case, you update the rom parameters locally but you're not saving them again as a json anymore. Is that alright?

SADPR replied:
Yes, I think you are not rewriting the RomParameters file. I am not sure if this fits the new workflow, but I guess you are just missing the following lines:
```
with open('RomParameters.json','w') as f:
            json.dump(updated_rom_parameters, f, indent = 4)
```

Rbravo555 replied:
You are right. Its been added

Rbravo555 committed:
Updating HROM test

Rbravo555 committed:
fixing bug with HROM

rubenzorrilla commented on applications/RomApplication/python_scripts/calculate_rom_basis_output_process.py:
```diff
@@ -58,10 +58,9 @@ def __init__(self, model, settings):
         self.rom_basis_output_format = settings["rom_basis_output_format"].GetString()
-        if not self.rom_basis_output_format == "json":
-            err_msg = "Provided \'rom_basis_output_format\' is {}. Available options are \'json\'.".format(self.rom_basis_output_format)
+        if not self.rom_basis_output_format == "json" and not self.rom_basis_output_format == "numpy" :
```
```suggestion
        rom_basis_output_available_formats = ["json", "numpy"]
        if self.rom_basis_output_format not in rom_basis_output_available_formats:
```
Just in case we add more in the future.

rubenzorrilla commented on applications/RomApplication/python_scripts/calculate_rom_basis_output_process.py:
```diff
@@ -136,6 +135,9 @@ def ExecuteFinalize(self):
         n_nodes = self.model_part.NumberOfNodes()
+        n_elements = self.model_part.NumberOfElements()
+        rom_basis_dict["hrom_settings"]["original_number_of_elements"] = n_elements
+        rom_basis_dict["hrom_settings"]["hrom_format"] = self.rom_basis_output_format
```
```suggestion
        rom_basis_dict["hrom_settings"]["output_format"] = self.rom_basis_output_format
```
I think `hrom` is redundant in this case. Besides, I also think that `output_format` is more self-descriptive.

rubenzorrilla commented on applications/RomApplication/python_scripts/calculate_rom_basis_output_process.py:
```diff
@@ -151,20 +153,27 @@ def ExecuteFinalize(self):
             raise Exception(err_msg)
 
+        # Creating the RomParameters.json containing or not the modes depending on "self.rom_basis_output_format"
+        output_filename = "RomParameters.json"
```
Are we 100% sure we want to hardcode this? I think it might be handy to keep the optional naming (OFC defaulting to `RomParameters.json`).

SADPR replied:
As previously mentioned in my comment regarding this matter, I concur with Ruben's viewpoint. I further suggested that a future pull request should include the feature of allowing the user to pass the name of the "RomParameters" to the ``RomAnalysis`` function.

Rbravo555 replied:
The issue here is, even when the field "rom_basis_output_name" seemed to provide the user the freedom to choose the name of the file to store the ROM information; in reality, the name RomPameters is hardcoded elsewhere within the workflow. Therefore, choosing a different name here, will invariably lead to a FILE_NOT_FOUND error.

rubenzorrilla commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -41,6 +41,7 @@ def __init__(self, solver, custom_settings):
         self.rom_settings = custom_settings["rom_settings"]
         self.hrom_visualization_model_part = settings["create_hrom_visualization_model_part"].GetBool()
         self.projection_strategy = settings["projection_strategy"].GetString()
+        self.hrom_output_format = settings["hrom_format"].GetString()
```
```suggestion
        self.hrom_output_format = settings["output_format"].GetString()
```
Same comment.

rubenzorrilla commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -132,6 +134,8 @@ def CreateHRomModelParts(self):
     @classmethod
     def __GetHRomTrainingDefaultSettings(cls):
         default_settings = KratosMultiphysics.Parameters("""{
+            "original_number_of_elements": 10,
```
```suggestion
            "original_number_of_elements": 0,
```
I'd default this to zero to then check that a valid value is provided, in other words, if after the defaults check the `original_number_of_elements` is still 0 we throw an error.

rubenzorrilla commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -132,6 +134,8 @@ def CreateHRomModelParts(self):
     def __GetHRomTrainingDefaultSettings(cls):
         default_settings = KratosMultiphysics.Parameters("""{
+            "original_number_of_elements": 10,
+            "hrom_format": "numpy",
```
```suggestion
            "output_format": "numpy",
```

rubenzorrilla commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -206,19 +195,67 @@ def __AppendHRomWeightsToRomParameters(self):
 
         if self.echo_level > 0 : KratosMultiphysics.Logger.PrintInfo("HRomTrainingUtility","\'RomParameters.json\' file updated with HROM weights.")
+
+
```
```suggestion
```
Remove useless lines.

rubenzorrilla commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -206,19 +195,67 @@ def __AppendHRomWeightsToRomParameters(self):
+
+        return updated_weights, updated_elements
+
+
```
```suggestion
```

rubenzorrilla commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -206,19 +195,67 @@ def __AppendHRomWeightsToRomParameters(self):
+        updated_weights = np.r_[original_weights, np.zeros(added_conditions_ids.shape)]
+
+        return updated_weights, updated_conditions
+
```
```suggestion
```

rubenzorrilla commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -206,19 +195,67 @@ def __AppendHRomWeightsToRomParameters(self):
+        if w is None:
+            w = np.load('WeightsMatrix.npy')
+        if z is None:
+            z = np.load('ElementsVector.npy')
```
As far as I understand, in here you are assuming the new numpy output formatting. Would the json-based one still work with this?

Rbravo555 replied:
JSON format will still work, and all the ROM info will be stored in the RomParameters.json if the flag is set for "json" rather than "numpy"

rubenzorrilla commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -206,19 +195,67 @@ def __AppendHRomWeightsToRomParameters(self):
+                hrom_weights["Conditions"][int(z)-n_elements] = float(w)
+        else:
+            # Many elements found
+            for j in range (0,len(z)):
```
```suggestion
            for j in range (len(z)):
```

rubenzorrilla commented on applications/RomApplication/python_scripts/rom_analysis.py:
```diff
@@ -110,26 +111,47 @@ def ModifyAfterSolverInitialize(self):
-                petrov_galerkin_nodal_dofs = len(self.project_parameters["solver_settings"]["rom_settings"]["nodal_unknowns"].GetStringArray())
-                aux = KratosMultiphysics.Matrix(petrov_galerkin_nodal_dofs, self.petrov_galerkin_rom_dofs)
+            # Set the right nodal ROM basis
+            if self.rom_parameters["rom_format"].GetString() == "json":
```
```suggestion
            if self.rom_parameters["output_format"].GetString() == "json":
```
Same comment.

Now I realize that you might have done it this way (`rom_format` and `hrom_format`) to allow for different formats for the modes and the weights. If so, I'd make a unique format for both for the sake of simplicity.

rubenzorrilla commented on applications/RomApplication/python_scripts/rom_analysis.py:
```diff
@@ -110,26 +111,47 @@ def ModifyAfterSolverInitialize(self):
+            elif self.rom_parameters["rom_format"].GetString() == "numpy":
+                #### Set the nodal ROM basis ####
+                Nodes = np.load("NodeIds.npy")
+                RightModes = np.load("RightBasisMatrix.npy")
```
Maybe not for this PR, but I think that we could consider to make these filenames user-definable as it might be handy in some situations.

Rbravo555 replied:
A future PR should adress the naming of the RomParameters.json and all the numpy arrays.

rubenzorrilla commented on applications/RomApplication/python_scripts/rom_analysis.py:
```diff
@@ -110,26 +111,47 @@ def ModifyAfterSolverInitialize(self):
+            elif self.rom_parameters["rom_format"].GetString() == "numpy":
+                #### Set the nodal ROM basis ####
+                Nodes = np.load("NodeIds.npy")
+                RightModes = np.load("RightBasisMatrix.npy")
```
```suggestion
                nodes = np.load("NodeIds.npy")
                right_modes = np.load("RightBasisMatrix.npy")
```
Please stick to our [style guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide). In this case, note that a local variable is not written in `snake_case` as it should be but in `CamelCase`.

rubenzorrilla commented on applications/RomApplication/python_scripts/rom_analysis.py:
```diff
@@ -110,26 +111,47 @@ def ModifyAfterSolverInitialize(self):
+                RightModes = RightModes[:,:rom_dofs]
+                if (self.solving_strategy == "petrov_galerkin"):
+                    petrov_galerkin_rom_dofs = self.project_parameters["solver_settings"]["rom_settings"]["petrov_galerkin_number_of_rom_dofs"].GetInt()
+                    LeftModes = np.load("LeftBasisMatrix.npy")
```
```suggestion
                    left_modes = np.load("LeftBasisMatrix.npy")
```
Same style guide comment.

rubenzorrilla commented on applications/RomApplication/python_scripts/rom_analysis.py:
```diff
@@ -138,14 +160,28 @@ def ModifyAfterSolverInitialize(self):
+                    # Set the HROM weights in elements and conditions
+                    hrom_weights_elements = self.rom_parameters["elements_and_weights"]["Elements"]
+                    for key,value in zip(hrom_weights_elements.keys(), hrom_weights_elements.values()):
+                        print('value being imposed to element ',key , '  is:', value.GetDouble())
```
Isn't this a debugging leftover?
```suggestion
```

Rbravo555 replied:
It was a leftover, it's been cleaned

rubenzorrilla commented on applications/RomApplication/python_scripts/rom_analysis.py:
```diff
@@ -138,14 +160,28 @@ def ModifyAfterSolverInitialize(self):
+                    hrom_weights_condtions = self.rom_parameters["elements_and_weights"]["Conditions"]
+                    for key,value in zip(hrom_weights_condtions.keys(), hrom_weights_condtions.values()):
+                        print('value being imposed to condition ',key , ' is:', value.GetDouble())
+
```
```suggestion
```
Same.

rubenzorrilla commented on applications/RomApplication/python_scripts/rom_analysis.py:
```diff
@@ -138,14 +160,28 @@ def ModifyAfterSolverInitialize(self):
+                    # Set the HROM weights in elements and conditions
+                    WeightsMatrix = np.load("WeightsMatrix.npy")
+                    ElementsVector = np.load("ElementsVector.npy")
+                    OriginalNumberOfElements = self.rom_parameters["hrom_settings"]["original_number_of_elements"].GetInt()
```
```suggestion
                    weights_matrix = np.load("WeightsMatrix.npy")
                    elements_vector = np.load("ElementsVector.npy")
                    original_number_of_elements = self.rom_parameters["hrom_settings"]["original_number_of_elements"].GetInt()
```

rubenzorrilla replied:
Same style guide comment.

rubenzorrilla commented on applications/RomApplication/tests/test_calculate_rom_basis_output_process.py:
```diff
@@ -33,7 +33,6 @@ def testCalculateRomBasisOutputProcess(self):
             "snapshots_interval": 1.0,
             "nodal_unknowns": ["TEMPERATURE"],
             "rom_basis_output_format": "json",
-            "rom_basis_output_name": "RomParameters",
```
Same comment. I think we should keep this user-definable.

Rbravo555 replied:
The issue here is, even when the field "rom_basis_output_name" seemed to provide the user the freedom to choose the name of the file to store the ROM information; in reality, the name RomPameters is hardcoded elsewhere within the workflow. Therefore, choosing a different name here, will invariably lead to a FILE_NOT_FOUND error.

In a future PR, we shall allow to name the file differently, which might come handy for "non-standard" simulations like coupled problems. Of course the solution shall involve other sections of the code and probably should happen within the RomManager #10763

rubenzorrilla commented on applications/RomApplication/tests/test_compressible_potiential_rom.py:
```diff
@@ -1,5 +1,5 @@
 import os
-
+import numpy as np
```
Why is this required if not used in this file?

Rbravo555 replied:
For that particular test it was required but not being imported. The failing test did not trigger any alarm because the RomApplication does not apper in the Continuous Integration any longer

rubenzorrilla replied:
OK, but didn't we add the `RomApplication` to the CI at some point?

Rbravo555 replied:
We just spoke with @roigcarlo and found out that the RomApplication is indeed still in the CI. It just so happens that the "nightly" test suite is empty. We only have the tests in the "small" one. We shall add it in a future breve PR

rubenzorrilla reviewed (commented):
Some minor comments. Aside of these, it looks good to me.

Rbravo555 committed:
Erasing debugging prints

Rbravo555 committed:
Update applications/RomApplication/python_scripts/calculate_rom_basis_output_process.py

Co-authored-by: Rubén Zorrilla <rubenzorrillamartinez@hotmail.com>

Rbravo555 committed:
Update applications/RomApplication/python_scripts/hrom_training_utility.py

Co-authored-by: Rubén Zorrilla <rubenzorrillamartinez@hotmail.com>

Rbravo555 committed:
Update applications/RomApplication/python_scripts/hrom_training_utility.py

Co-authored-by: Rubén Zorrilla <rubenzorrillamartinez@hotmail.com>

Rbravo555 committed:
Update applications/RomApplication/python_scripts/hrom_training_utility.py

Co-authored-by: Rubén Zorrilla <rubenzorrillamartinez@hotmail.com>

Rbravo555 committed:
Update applications/RomApplication/python_scripts/rom_analysis.py

Co-authored-by: Rubén Zorrilla <rubenzorrillamartinez@hotmail.com>

Rbravo555 committed:
Update applications/RomApplication/python_scripts/rom_analysis.py

Co-authored-by: Rubén Zorrilla <rubenzorrillamartinez@hotmail.com>

Rbravo555 committed:
Update applications/RomApplication/python_scripts/hrom_training_utility.py

Co-authored-by: Rubén Zorrilla <rubenzorrillamartinez@hotmail.com>

Rbravo555 committed:
Update applications/RomApplication/python_scripts/hrom_training_utility.py

Co-authored-by: Rubén Zorrilla <rubenzorrillamartinez@hotmail.com>

Rbravo555 committed:
Update applications/RomApplication/python_scripts/rom_analysis.py

Co-authored-by: Rubén Zorrilla <rubenzorrillamartinez@hotmail.com>

Rbravo555 committed:
style update

Rbravo555 committed:
including conditions' indexes and weights as extra numpy files

Rbravo555 committed:
accepting conditions' indexes and weights

Rbravo555 committed:
eliminating number of elements from RomParameters.json

SADPR commented on applications/RomApplication/python_scripts/calculate_rom_basis_output_process.py:
```diff
@@ -58,10 +58,10 @@ def __init__(self, model, settings):
+        if self.rom_basis_output_format not in rom_basis_output_available_formats:
+            err_msg = "Provided \'rom_basis_output_format\' is {}. Available options are \'json\' and \'numpy\'.".format(self.rom_basis_output_format)
             raise Exception(err_msg)
-        self.rom_basis_output_name = settings["rom_basis_output_name"].GetString()
```
In my opinion, it would be beneficial to retain the choice of "rom_basis_output_name". For example, in the context of CoSimulation for fluid-thermal coupling, it is necessary to have separate files named RomParameters_fluid.json and RomParameters_thermal.json. Therefore, I propose allowing the ability to pass the name of the "RomParameters" file to the ``RomAnalysis`` (this can be done in a future PR).

SADPR commented on applications/RomApplication/python_scripts/calculate_rom_basis_output_process.py:
```diff
@@ -151,20 +152,27 @@ def ExecuteFinalize(self):
+        elif self.rom_basis_output_format == "numpy":
+            # Storing modes in Numpy format
+            numpy.save('RightBasisMatrix.npy', u)
+            numpy.save('NodeIds.npy',  numpy.arange(1,((u.shape[0]+1)/n_nodal_unknowns), 1, dtype=int)   )
```
I am curious about whether this would still work if there is a floating node within the model part, although this is highly unlikely. To provide an example, suppose there is a "FluidModelPart" that contains points that were generated from CAD and are not included in the computational model part.

Rbravo555 replied:
If that specific case does not work, it would not work for the JSON format either

SADPR commented on applications/RomApplication/python_scripts/calculate_rom_basis_output_process.py:
```diff
@@ -151,20 +152,27 @@ def ExecuteFinalize(self):
         rom_basis_dict["rom_settings"]["number_of_rom_dofs"] = numpy.shape(u)[1] #TODO: This is way misleading. I'd call it number_of_basis_modes or number_of_rom_modes
         rom_basis_dict["projection_strategy"] = "galerkin" # Galerkin: (Phi.T@K@Phi dq= Phi.T@b), LSPG = (K@Phi dq= b), Petrov-Galerkin = (Psi.T@K@Phi dq = Psi.T@b)
```
```suggestion
        rom_basis_dict["projection_strategy"] = "galerkin" # Galerkin: (Phi.T@K@Phi dq= Phi.T@b), LSPG = (K@Phi dq= b), Petrov-Galerkin = (Psi.T@K@Phi dq = Psi.T@b)
        rom_basis_dict["rom_format"] = self.rom_basis_output_format
```

Rbravo555 committed:
Adapting HROM test

Rbravo555 committed:
Update applications/RomApplication/python_scripts/calculate_rom_basis_output_process.py

Co-authored-by: Sebastian Ares de Parga Regalado <61457043+SADPR@users.noreply.github.com>

SADPR commented on applications/RomApplication/python_scripts/hrom_training_utility.py:
```diff
@@ -156,28 +159,12 @@ def __CalculateResidualBasis(self):
-                else:
-                    hrom_weights["Conditions"][int(z[j])-n_elements] = float(w[j])
+        # Create dictionary with HROM weights (Only used for the expansion of the selected Conditions to include their parent Elements)
+        hrom_weights = self.__CreateDictionaryWithRomElementsAndWeights(weights,indexes,number_of_elements)
```
This function is called regardless of the format type ("json" or "numpy"). Inside this function, it tries to read numpy files that may not exist if the user is training with "json".

SADPR replied:
Hyper-reduction training does not work for "json".

Rbravo555 replied:
Now it should work

Rbravo555 committed:
Adding numpy to training of Petrov-Galerkin

Rbravo555 committed:
Storing updated JSON if flag is active

Rbravo555 committed:
erase spaces

Rbravo555 committed:
Add "numpy" to Petrov-Galerkin test

Rbravo555 committed:
Adding back the option to change the name of the JSON file of the ROM

Rbravo555 committed:
Changing test of ROM basis output process

SADPR commented on applications/RomApplication/tests/structural_static_test_files/HROM/RomParameters.json:
```diff
@@ -14,7 +15,10 @@
-    "hrom_settings": {},
+    "hrom_settings": {
+        "hrom_format":"numpy",
+        "original_number_of_elements":116
```
Should this be here?

Rbravo555 committed:
erasing unused "original number of elements"

SADPR commented on applications/RomApplication/tests/test_calculate_rom_basis_output_process.py:
```diff
@@ -33,7 +33,6 @@ def testCalculateRomBasisOutputProcess(self):
             "snapshots_interval": 1.0,
             "nodal_unknowns": ["TEMPERATURE"],
             "rom_basis_output_format": "json",
```
```suggestion
            "rom_basis_output_format": "json",
            "rom_basis_output_name": "RomParameters",
```

Rbravo555 committed:
Adding "RomParameters" as name for test rom basis output process

SADPR reviewed (commented):
Two comments away from approval.

SADPR reviewed (approved):
All set. Ready.
